code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -32,6 +35,21 @@
 		icon_state = "glass_empty"
 		renamedByPlayer = FALSE //so new drinks can rename the glass
 
+/obj/item/reagent_containers/food/drinks/drinkingglass/AltClick(mob/user)
+	. = ..()
+	if(length(reagents.reagent_list) && !flipped)
+		to_chat(user, span_danger("You probably shouldn't flip this over with something in it!"))
+		return
+	flipped = !flipped
+	playsound(src, drop_sound, DROP_SOUND_VOLUME, vary = sound_vary, ignore_walls = FALSE)
+	if(flipped)
+		icon_state = initial(icon_state) + "_flipped"
```

Comment:
Why have new icon states instead of rotating the icon state.

Reply:
Because the glasses have perspective